content/building-a-simple-keras-and-deep-learning-rest-api.md

```diff
@@ -0,0 +1,390 @@
+
+Specifically, we will learn:
+
+- How to (and how not to) load a Keras model into memory so it can be efficiently used for inference
+- How to use the Flask web framework to create an endpoint for our API
+- How to make predictions using our model, JSON-ify them, and return the results to the client
+- How to call our Keras REST API using both cURL and Python
+
+By the end of this tutorial you'll have a good understanding of the components (in their simplest form) that go into a creating Keras REST API.
+
+Feel free to use the code presented in this guide as a starting point for your own deep learning REST API.
+
+**Note: The method covered here is intended to be instructional. It is _not_ meant to be production-level and capable of scaling under heavy load. If you're interested in a more advanced Keras REST API that leverages message queues and batching, [please refer to this tutorial](https://www.pyimagesearch.com/2018/01/29/scalable-keras-deep-learning-rest-api/).**
+
+---
+
+## Configuring your development environment
+
+We'll be making the assumption that Keras is already configured and installed on your machine. If not, please ensure you install Keras using the [official install instructions](https://keras.io/#installation).
+
+From there, we'll need to install [Flask](http://flask.pocoo.org/) (and its associated dependencies), a Python web framework, so we can build our API endpoint. We'll also need [requests](http://docs.python-requests.org/en/master/) so we can consume our API as well.
+
+The relevant `pip` install commands are listed below:
+
+```sh
+$ pip install flask gevent requests
+```
+
+---
+
+## Building your Keras REST API
+
+Our Keras REST API is self-contained in a single file named `run_keras_server.py`. We kept the installation in a single file as a manner of simplicity &mdash; the implementation can be easily modularized as well.
+
+Inside `run_keras_server.py` you'll find three functions, namely:
+
+- `load_model`: Used to load our trained Keras model and prepare it for inference.
+- `prepare_image`: This function preprocesses an input image prior to passing it through our network for prediction. If you are not working with image data you may want to consider changing the name to a more generic `prepare_datapoint` and applying any scaling/normalization you may need.
+- `predict`: The actual endpoint of our API that will classify the incoming data from the request and return the results to the client.
+
+The full code to this tutorial can be found [here](https://github.com/jrosebr1/simple-keras-rest-api).
+
+```python
+# import the necessary packages
+from keras.applications import ResNet50
+from keras.preprocessing.image import img_to_array
+from keras.applications import imagenet_utils
+from flask import Flask
+from flask import jsonify
+from flask import request
```

Comment:
I would recommend `import flask` then using e.g. `flask.request`, `flask.jsonify`. Especially in order to avoid confusion with `import requests` later on.

Reply:
Great point on avoiding confusion. Updating imports and associated code.